File: Lift.Buddy.Core/Database/Entities/WorkoutPlan.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Lift.Buddy.Core.Database.Entities;

public class WorkoutPlan
{
    public WorkoutPlan()
    {
        Users = new HashSet<User>();
        WorkoutDays = new HashSet<WorkoutDay>();
    }

    public Guid WorkoutPlanId { get; set; }

    public string Name { get; set; }
    public double ReviewAverage { get; set; }
    public Guid CreatorId { get; set; }
    public DateTime CreationDate { get; set; }
    public virtual User Creator { get; set; }
    public virtual ICollection<Review> Reviews { get; set; }
    public virtual ICollection<User> Users { get; set; }
    public virtual ICollection<WorkoutDay> WorkoutDays { get; set; }
}

File: UI/ClientApp/src/app/Model/WorkoutPlan.ts
import { WorkoutDay } from "./WorkoutDay";

export class WorkoutPlan {
    public id: string = '';
    public name: string = '';
    public createdBy: string = '';
    public workoutDays: WorkoutDay[] = [];
    public reviewAverage: number = 0;
    public myReview: number = 0;
}
